airflow/jobs/scheduler_job.py

```diff
@@ -1478,6 +1478,7 @@ def _fail_tasks_stuck_in_queued(self, session: Session = NEW_SESSION) -> None:
                             self._task_queued_timeout,
                         )
                         ti.handle_failure(error=msg, session=session)
+                        self.executor.cleanup_stuck_queued_task(ti)
```

Comment:
Do we really need the ti.handle_failure? Won't it be handled in `Scheduler._process_executor_events`?

Also, I think we shouldn't iterate the task instances here but send all to the executor, then have the executor iterate and remove them from the running. When we iterate here, we call the executor cleanup several times but we could call it once with the task instances and then have it remove them from the running tasks. Looks more efficient _I think_